src/impl/load.c

```diff
@@ -0,0 +1,289 @@
+            
+            game_id_str = malloc_or_die(id_len + 1);
+            strncpy(game_id_str, identifier + start, id_len);
+            game_id_str[id_len] = '\0';
+        } else {
+            regfree(&regex);
+            error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                            get_formatted_string("Failed to extract game ID from woogles URL: %s", identifier));
+            return NULL;
+        }
+
+        regfree(&regex);
+    } else {
+        // Check if it's a standalone woogles game ID (must be alphanumeric only)
+        if (is_all_digits_or_empty(identifier)) {
+            // All digits likely means cross-tables, not woogles
+            return NULL;
+        }
+        // Check if identifier is alphanumeric only
+        for (int i = 0; identifier[i]; i++) {
+            if (!isalnum(identifier[i])) {
+                // Contains non-alphanumeric characters, not a woogles game ID
+                return NULL;
+            }
+        }
+        // It's alphanumeric and not all digits, assume it's a woogles game ID
+        game_id_str = string_duplicate(identifier);
+    }
+    
+    char *gcg_content = NULL;
+
+    if (game_id_str) {
+        // Use woogles API to get GCG content
+        char *curl_cmd = get_formatted_string("curl -s -H 'Content-Type: application/json' "
+                                              "https://woogles.io/api/game_service.GameMetadataService/GetGCG "
+                                              "-d '{\"game_id\":\"%s\"}'", game_id_str);
+        
+        char *response = get_process_output(curl_cmd);
+        free(curl_cmd);
+        free(game_id_str);
+        
+        if (response) {
+            // Parse JSON response to extract GCG content
+            const char *gcg_start = strstr(response, "\"gcg\":\"");
+            if (gcg_start) {
+                gcg_start += 7; // Skip "\"gcg\":\""
+                const char *gcg_end = gcg_start;
+                while (*gcg_end && *gcg_end != '"') {
+                    gcg_end++;
+                }
+                int gcg_len = gcg_end - gcg_start;
+                char *raw_gcg = malloc_or_die(gcg_len + 1);
+                strncpy(raw_gcg, gcg_start, gcg_len);
+                raw_gcg[gcg_len] = '\0';
+                
+                // Unescape JSON string using utility function
+                gcg_content = json_unescape_string(raw_gcg, error_stack);
+                free(raw_gcg);
+                
+                if (!gcg_content) {
+                    free(response);
+                    return NULL;
+                }
+            }
+            
+            free(response);
+        }
+    }
+    
+    if (!gcg_content || is_string_empty_or_whitespace(gcg_content) ||
+        strstr(gcg_content, "#player1") == NULL) {
+        free(gcg_content);
+        // Only push error if this was actually a woogles URL
+        if (strstr(identifier, "woogles.io") != NULL) {
```

Comment:
Same as above, we should return an error for bad gcg_content for any identifier.